st3/sublime_lib/fancy_settings.py

```diff
@@ -0,0 +1,47 @@
+class FancySettings():
+    def __init__(self, settings):
+        self.settings = settings
+
+    def get(self, name, default=None):
+        return self.settings.get(name, default)
+
+    def set(self, name, value):
+        self.settings.set(name, value)
+
+    def erase(self, name):
+        self.settings.erase(name)
+
+    def has(self, name):
+        return self.settings.has(name)
+
+    def __getitem__(self, key):
+        return self.get(key)
+
+    def __setitem__(self, key, value):
+        self.set(key, value)
+
+    def __delitem__(self, key):
+        self.erase(key)
+
+    def __contains__(self, item):
+        return self.has(item)
+
+    def subscribe(self, key, selector, callback):
```

Comment:
What is the purpose of `key` here? Imho we should try to take the `add_on_change` key away from the user so they don't have to bother with it. Besides, if you fall back the key to `selector` when it's a string, it quickly becomes non-unique.

Reply:
`key` is there solely so that users can unsubscribe. Setting `key` when `selector` was a string was a mistake; I've fixed it.

Reply:
That is a much better idea.